tests/phpunit/CRM/RcBase/Api/CreateHeadlessTest.php

```diff
@@ -0,0 +1,365 @@
+<?php
+
+use Civi\Test\HeadlessInterface;
+use PHPUnit\Framework\TestCase;
+
+/**
+ * Test API Create class
+ *
+ * @group headless
+ */
+class CRM_RcBase_Api_CreateHeadlessTest extends TestCase implements HeadlessInterface
+{
+    /**
+     * Test contact ID
+     *
+     * @var int
+     */
+    private $testContactId;
+
+    /**
+     * The setupHeadless function runs at the start of each test case, right before
+     * the headless environment reboots.
+     *
+     * It should perform any necessary steps required for putting the database
+     * in a consistent baseline -- such as loading schema and extensions.
+     *
+     * The utility `\Civi\Test::headless()` provides a number of helper functions
+     * for managing this setup, and it includes optimizations to avoid redundant
+     * setup work.
+     *
+     * @see \Civi\Test
+     */
+    public function setUpHeadless()
+    {
+        return \Civi\Test::headless()
+            ->installMe(__DIR__)
+            ->apply();
+    }
+
+    /**
+     * Create a clean DB before running tests
+     *
+     * @throws CRM_Extension_Exception_ParseException
+     */
+    public static function setUpBeforeClass(): void
```

Comment:
When i generate the test files with civix command, this function is not generated. What is it for?

Reply:
It's executes once before the tests and resets the DB.
```
        \Civi\Test::headless()
            ->installMe(__DIR__)
            ->apply(true);
```
Here the `true` param for `apply()` forces a new build for the DB.